### ISSUE TITLE
Accessibility? 

### ISSUE THREAD

**Issue by medeaoblongata** [colinmeinke/react-svg-chart#9]
Your SVG views can be made much more accessible quite easily by adding a little ARIA.

Check https://www.w3.org/TR/graphics-aria-1.0/ for the basic Graphics ARIA spec. (You can also inspect [the full ARIA spec](https://www.w3.org/TR/wai-aria-1.2/) - I suggest paying close attention to the notion of roles, the importance of naming and the spec for `aria-roledescription`).

Appropriate role and role description for **chart** :

`<svg role="graphics-document" aria-roledescription="line chart">...` (use the chart **type** for the role description. The value should be available for internationalisation, because it is intended for a human audience).

Rely on SVG `<title>` for the chart name. This has broad support across user agents and assistive tech. 
I humbly suggest that your API should **require** a non-null title for every chart. (Change my mind).

Appropriate role and role description and label for **axes** and **legends** :
Use graphics-object for stuff that contains other meaningful content (e.g. an axis that contains axis values or a subset of datapoints).
`<g role="graphics-object" aria-roledescription="x axis">...`

Appropriate role and role description and label for **datapoints and axis values**.
Use graphics-symbol for stuff that is meaningful in itself, but contains no additional meaningful 'children'.
`<circle role="graphics-symbol" aria-label="30"...` bar chart does not use circle but the concept is the same even with `<g>`

(Note that all children of "graphics-symbol" are treated as 'presentational', and will be excluded from the accessibility tree).

SVG `<text>` elements can be included in the accessibility tree if the roles are set up properly. 

You can also do something like 
`<g role="graphics-object" aria-labelledby="id_of_the_onscreen_text">`

A datapoint may be associated with a particular legend element using `aria-describedby`. 

The legend can even be in HTML, if that is more appropriate. (DL/DT/DD offers a compellingly correct semantic, although there are also viable ARIA equivalents if you prefer SVG). `<foreignObject>` can handle HTML inside SVG quite well, and reflows beautifully across a wide range of browsers, but be careful about the role you use.

"Visual affordances" such as axis tickmarks or grid lines are "purely presentational". They will be excluded from the accessibility tree unless you give them a name/role. Screen reader users do not welcome redundant chatter.

Finally, a small caveat. Make sure that your library outputs the datapoints to the SVG subtree in the same order that they appear on screen. It does **not** need to be left-to-right, or top-to-bottom. It could even be a circular arrangement (such as a piechart). The important thing is that the visual sequence should match the DOM sequence. AFAICT this already happens. WCAG violations are likely if you deviate from this approach.

These suggestions will not guarantee WCAG compliance but they will make your charts much more accessible, without affecting the visual appearance.

Good luck!
